File: program/convert-string-to-vowelcase/ConvertStringToVowelcase.rb
module Vowelcase

  VOWELS = %w[a e i o u]
  
  def self.vowel(string)
    result = ""
    string.chars { |c| VOWELS.include?(c) ? result += c.upcase : result += c.downcase }
    result
  end
end

puts Vowelcase.vowel("hello world") # => hEllO wOrld

File: program/convert-string-to-vowelcase/convert_string_to_vowelcase.php
<?php
	function vowelCase($str) {
	  $vowels = ['a','e','i','o','u'];
	  $res = '';
	  for($i=0; $i< strlen($str); $i++){
	    if (in_array(strtolower($str[$i]), $vowels)){
	      $res .= strtoupper($str[$i]);
	    } else {
	      $res .= strtolower($str[$i]);
	    }
	  }
	  return $res;
	}
	
echo vowelCase('hello world');
?>
